documentation/projects/proposals/trust_and_safety/content_report_moderation/20231208-implementation_plan_django_admin_moderator_access.md

```diff
@@ -0,0 +1,587 @@
+- A chronological list of all moderation decisions for the work
+- A chronological list of all reports for the work regardless of status
+  - Include at least the following columns: status, description, reason, date
+
+The idea here is that the "report view" is not focused on taking an action for a
+single _report_, but rather showing all relevant information for a single work
+to make a moderation decision on the work, based on all pending and historical
+reports.
+
+We could take two approaches for this view: either create a brand new view or
+build this into the media view itself. My preference and recommendation is the
+latter, because while it creates complexity in the media admin view, that view
+doesn't currently have much of a purpose anyway, and if we're making it possible
+to view all the metadata for a work, we might as well take advantage of the
+existing admin view for media items that already has some of that built into it.
+I'll cover more details about how we'll accomplish this in the step-by-step.
+
+##### Actions
+
+To understand the rationale in this section, keep in mind that there are
+currently three report types (sensitive, DMCA, and "other"), and that users can
+select any of these. There is little guidance beyond the report type labels
+themselves to indicate to users which one to select. This means that reports of
+different types get mixed up, and especially the "other" type ends up with
+reports that would have been better suited as sensitive or DMCA reports. This
+means there is no coherent way to handle "all pending reports". A previous
+version of this implementation plan attempted to do that, but it's far too
+complex and has so many edge cases, that it's _easier_ to handle a set number of
+reports at a time, selected by the moderator.
+
+In the future, we could try to include links that open tables to show all
+moderation decisions for works by the same creator, etc, in case there is useful
+information, particularly for identifying potential bulk moderation actions.
+These kinds of additional views are out of scope for this current project,
+however.
+
+Each pending report should have a checkbox allowing it to be selected for
+actioning. If there is only a single pending report, it should be automatically
+selected when the form loads.
+
+The expanded view should have the following context-sensitive actions:
+
+- "Mark sensitive: mark the work as sensitive based on selected reports"
+  - Only available if a work is not already marked sensitive
+  - `confirmed_sensitive` action
+- "Deindex sensitive: entirely exclude the work from Openverse due to
+  sensitivity based on selected reports"
+  - There are no real conditions for this because deindexing can happen for
+    works that are or are not marked sensitive
+  - `deindexed_sensitive` action
+- "Deindex copyright: entirely exclude the work from Openverse due to copyright
+  based on selected reports"
+  - `deindexed_copyright`
```

Comment:
Should there be guidance for what to do if a record ought to be indexed for _both_ copyright and sensitivity concerns?

I think there's value in the separation especially if a work is deindexed without an official report (for example if it was the result of a bulk action), because the reason is preserved. And I suppose that if there are reports of multiple types that lead to an indexing decision, they're all linked to the ModerationDecision, so that context is not missed.

Reply:
Hmm, that's interesting. I really don't know, other than either adding a list field that includes the reasons (so like, the action would be `deindexed` and then the moderator would select the reasons as both sensitivity and copyright)? Or maybe a `deindexed_sensitive_and_copyright`? To me, I feel like copyright nullifies the sensitivity question.

> And I suppose that if there are reports of multiple types that lead to an indexing decision, they're all linked to the ModerationDecision, so that context is not missed.

By this, do you mean to say that you feel the context is already sufficiently covered?

This case, where we need to know that it was both for sensitivity and copyright, would only be an issue if we're overturning deindex decisions based on appeals at some future date, right? Those appeals would be manually reviewed, and the moderator would presumably look at the moderation decision history then. But for bulk actions or if moderators are working quickly and not necessarily adding explanations, it could cause an issue.

What do you think? Is it worth increasing the complexity to capture the reason as a separate thing?

Reply:
I meant that I came to the same conclusion as you, I think -- the copyright nullifies the sensitivity question. So if a record has copyright and sensitivity reports, I think it would make sense for a moderator to use the `deindexed_copyright` action as that takes "priority", but link that ModerationDecision to _all_ the reports, including the sensitivity ones.

Then, when you're looking at the ModerationDecision in the future, the sensitivity reports will still be linked to it, so you'll still be able to see the full context.

Basically I think I talked myself into the same thing -- I think the context is covered sufficiently, and it's not worth adding more complexity 😄 Just reasoning through the edge cases!